File: Java/week3/PigLatin.java

/*
 Filename: PigLatin.java
 Author: Mikhail Zubko
 Created: 2025/01/27
 Purpose: 
*/
import javax.swing.JOptionPane;

public class PigLatin {
  public static void main(String[] args) {
    final String VOWELS = "aeiouAEIOU";
    String word = "";
    String pigword = "";
    while (true) {
      word = JOptionPane.showInputDialog(null, "Enter a word\nto be converted to Pig Latin", "Input a Word",
          JOptionPane.QUESTION_MESSAGE);
      if (word == null) {
        break; // cancelled
      }
      if (word.isBlank()) {
        continue; // empty input
      }
      if (VOWELS.indexOf(word.charAt(0)) != -1) {
        pigword = word + "way"; // if the word starts with vowel
      } else { // if the word starts with consonant
        for (int i = 0; i < word.length(); i++) {
          if (VOWELS.indexOf(word.charAt(i)) != -1) {
            // find the index of vowel and use it to split the word
            pigword = word.substring(i) + word.substring(0, i) + "ay";
            break;
          } else if (i == word.length() - 1) {
            pigword = word + "ay"; // if no vowels in the word
          }
        }
      }
      JOptionPane.showMessageDialog(null, "Pig Latin version:\n" + pigword);
    }
  }
}

File: Cpp/week12/piglatin/piglatin.cpp
#include <algorithm>
#include <iostream>
#include <string>

int main() {
  const std::string VOWELS = "aeiouAEIOU";
  std::string word;
  std::string pigword;
  while (true) {
    std::cout << "Enter a word\nto be converted to Pig Latin (q to exit): ";
    std::cin >> word;
    if (word == "q") {
      break;
    }
    if (std::find(VOWELS.begin(), VOWELS.end(), word[0]) != VOWELS.end()) {
      pigword = word + "way";
    } else {
      for (int i = 0; i < word.size(); i++) {
        if (std::find(VOWELS.begin(), VOWELS.end(), word[i]) != VOWELS.end()) {
          pigword = word.substr(i) + word.substr(0, i) + "ay";
          break;
        } else if (i == word.size() - 1) {
          pigword = word;
        }
      }
    }
    std::cout << "PigLatin word: " << pigword << std::endl;
  }

  return 0;
}
